scripts/loki_transform.py

```diff
@@ -467,39 +474,63 @@ def ecphys(mode, config, header, source, build, cpp, directive, frontend):
 
     # Remove DR_HOOK and other utility calls first, so they don't interfere with SCC loop hoisting
     if 'scc' in mode:
-        scheduler.process(transformation=RemoveCallsTransformation(
+        utility_call_trafo = RemoveCallsTransformation(
             routines=['DR_HOOK', 'ABOR1', 'WRITE(NULOUT'], include_intrinsics=True, kernel_only=True
-        ))
+        )
     else:
-        scheduler.process(transformation=DrHookTransformation(mode=mode, remove=False))
+        utility_call_trafo = DrHookTransformation(mode=mode, remove=False)
+    scheduler.process(transformation=utility_call_trafo)
 
-    # Now we instantiate our transformation pipeline and apply the main changes
+    # Instantiate transformation pipeline and apply the main changes
     transformation = None
     if mode == 'idem':
         transformation = IdemTransformation()
 
     if mode == 'sca':
         # Define the target dimension to strip from kernel and caller
-        horizontal = scheduler.config.dimensions['horizontal']
         transformation = ExtractSCATransformation(horizontal=horizontal)
```

Comment:
Do we not need to wrap this in a tuple? Same for the idem trafo above.

Reply:
Good catch! I've added an `as_tuple` around the transformations in the application loop, which will have the same effect, but makes it a little safer.